### ISSUE TITLE
core.playlists.create returns "null" playlist

### ISSUE THREAD

**Issue by tkem** [mopidy/mopidy#1162]
Mopidy v1.0.4:
```
"websocket:outgoingMessage" Object { method: "core.playlists.create", params: Object, jsonrpc: "2.0", id: 2 } localhost:8100:30:31
"websocket:incomingMessage" message { target: WebSocket, isTrusted: true, data: "{"jsonrpc": "2.0", "id": 2, "result": null}", origin: "ws://127.0.0.1:6680", lastEventId: "", eventPhase: 0, bubbles: false, cancelable: false, defaultPrevented: false, timeStamp: 1430845313203906, originalTarget: WebSocket } localhost:8100:30:31
"websocket:incomingMessage" message { target: WebSocket, isTrusted: true, data: "{"playlist": null, "event": "playlist_changed"}", origin: "ws://127.0.0.1:6680", lastEventId: "", eventPhase: 0, bubbles: false, cancelable: false, defaultPrevented: false, timeStamp: 1430845313207905, originalTarget: WebSocket } localhost:8100:30:31
"event:playlistChanged" Object { playlist: null } localhost:8100:30:31
"websocket:outgoingMessage" Object { method: "core.playlists.as_list", jsonrpc: "2.0", id: 3 }
```
EDIT: For debug output, see below.

**Comment by tkem** [comment]
Nevermind.

**Comment by tkem** [comment]
Sorry for the confusion...
Real debug output is:
```
DEBUG    2015-05-05 19:07:13,336 [3323:HttpServer] mopidy.http.handlers
  Received WebSocket message from 127.0.0.1: u'{"method":"core.playlists.create","params":{"name":"tkem"},"jsonrpc":"2.0","id":4}'
DEBUG    2015-05-05 19:07:13,337 [3323:MainThread] mopidy.listener
  Sending playlist_changed to CoreListener: {'playlist': None}
```
So a `playlist.create({name: "tkem"})` apparently returns a `null` playlist.

**Comment by adamcik** [comment]
So re-reading our docs we claim that if URI is none we will pick the "first" one that supports playlists. So that would likely be based on the order the entry points get loaded I would presume.

So looking at the code you would like hit this code path https://github.com/mopidy/mopidy/blob/develop/mopidy/core/playlists.py#L128 which is already marked as possibly fishy. I would suspect the "first" one ended up being RO playlist provider, where we need to be looking for the ones that support RW. 

I also suspect this code will fail if there is no backend with playlists :/

**Comment by tkem** [comment]
Ahem, sorry, could have thought of that myself, but it was a long day...
Removing the `spotify` extension (which reports a "Spotify login error: Needs a premium account", so it shouldn't have been active, anyway) fixes this.

**Comment by tkem** [comment]
However, this leads us back to the question whether "read-only" playlist providers should be supported (and marked as such) at all, or if it wouldn't be better to use the browse interface for RO playlists...

**Comment by jodal** [comment]
We still haven't decided on anything here, but using browse for RO playlists and only implementing the playlist provider for RW playlists is something that has been discussed. I can't remember any huge drawbacks, except that it's not something we can enforce until 2.0, but we can decide on which direction to go in and update extensions accordingly.

**Comment by tkem** [comment]
Sure, no need to rush. Still, wearing my "client developer hat", this has been bugging me since starting with playlist support for Mopidy Mobile, so I think it's time to define what we really mean when calling something a "playlist".

**Comment by tkem** [comment]
Focusing back on the original issue, for an intermediate fix, I wonder if a "null" playlist result could mean "try the next playlist provider"?

**Comment by adamcik** [comment]
That sounds like a reasonable workaround, and something for the 1.0.x branch. Dp you time for a quick PR or are you too busy with Mopidy-Mobile? :-)

**Comment by tkem** [comment]
Current list of priorities:
a) work
b) MoMo
c) new slotcar racecourse
d) see what I'll can do

;-)

**Comment by jodal** [comment]
The "first one" is fishy. It made sense back when we had a `BACKENDS` configuration that was a list you  could reorder yourself. It's been fishy and ready for rethinking since extension support and the new config system landed in Mopidy 0.14 two years ago.

**Comment by tkem** [comment]
As stated before by @adamcik, the best thing we'll probably come up with short-term is loop through back ends until one returns a valid playlist (and doesn't throw). Will provide a PR along these lines in the next couple of days. Would love to see this in 1.0.5, since otherwise MoMo's new "create playlist" feature might break on any Mopidy installation with Spotify enabled. .
